### PR TITLE
Dilpreet | Test | Fixed UI exceptions

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/1mm9697
https://app.clickup.com/t/1uakqx2
https://app.clickup.com/t/1uatr6a
https://app.clickup.com/t/1vy1eng
https://app.clickup.com/t/1xbzm6t
https://app.clickup.com/t/1xbzn1a
https://app.clickup.com/t/1xbznbg
https://app.clickup.com/t/1xbznvb
https://app.clickup.com/t/1xbzpzw

### CODE REVIEW

**Review by the-prateek-mishra — APPROVED** [review]

File: apps/web-giddh/src/app/material-ledger/ledger.component.ts [hunk 1]
```diff
@@ -1461,7 +1461,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
             const currentLedgerCategory = this.lc.activeAccount ? this.generalService.getAccountCategory(this.lc.activeAccount, this.lc.activeAccount.uniqueName) : '';
             // If current ledger is of income or expense category then send current ledger as stockAccountUniqueName. Only required for ledger.
             const accountUniqueName = (currentLedgerCategory === 'income' || currentLedgerCategory === 'expenses') ?
-                this.lc.activeAccount ? this.lc.activeAccount.uniqueName : '' :
+                this.lc.activeAccount ? this.lc.activeAccount?.uniqueName : '' :
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/ledger.component.ts]:
```suggestion
                this.lc.activeAccount ? this.lc.activeAccount.uniqueName : '' :
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 2]
```diff
@@ -1355,8 +1355,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                         }
                         if (this.currentVoucherFormDetails?.depositAllowed || (this.isPendingVoucherType && obj.accountDetails && obj.voucherDetails)) {
                             if (!this.isLastInvoiceCopied) {
-                                obj.accountDetails.name = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account.customerName) ? results[0].account.customerName : results[0].account.name;
-                                obj.voucherDetails.customerName = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account.customerName) ? results[0].account.customerName : results[0].account.name;
+                                obj.accountDetails.name = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account?.customerName) ? results[0].account?.customerName : results[0].account.name;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                                obj.accountDetails.name = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account?.customerName) ? results[0].account?.customerName : results[0].account?.name;
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 3]
```diff
@@ -5008,7 +5008,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                     if (taxTypeArr.indexOf(ta.taxType) > -1) {
                         salesEntryClass.isOtherTaxApplicable = true;
                         let otherTaxModal = new SalesOtherTaxesModal();
-                        otherTaxModal.appliedOtherTax = { name: ta.name, uniqueName: ta.uniqueName };
+                        otherTaxModal.appliedOtherTax = { name: ta?.name, uniqueName: ta?.uniqueName };
                         otherTaxModal.tcsCalculationMethod = ta.calculationMethod;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                        otherTaxModal.tcsCalculationMethod = ta?.calculationMethod;
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 4]
```diff
@@ -5025,8 +5025,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                             uniqueName: ta.uniqueName,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                            uniqueName: ta?.uniqueName,
```

**Review by ravinderwalkover — COMMENTED** [review]
Fix issues and then merge

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**